src/Controls/src/Build.Tasks/CompiledMarkupExtensions/StaticResourceExtension.cs

```diff
@@ -0,0 +1,179 @@
+                        yield break;
+                    } 
+                    //multiple resources in <Resources>
+                    else if (resourcesNode is ListNode lr) {
+                        foreach (var rn in lr.CollectionItems) {
+                            if (rn is IElementNode irn2
+                                && irn2.Properties.TryGetValue(XmlName.xKey, out INode xKeyNode2) 
+                                && context.Variables.ContainsKey(irn2) 
+                                && xKeyNode2 is ValueNode xKeyValueNode2
+                                && xKeyValueNode2.Value as string == keyValueNode.Value as string)
+                            {
+                                if (irn2.CollectionItems.Count == 1 && irn2.CollectionItems[0] is ValueNode vn2 && vn2.Value is string) {
+                                    foreach (var instruction in TryConvert(vn2, eNode, vardefref, module, context))
+                                        yield return instruction;
+                                    yield break;
+                                }
+
+                                vardefref.VariableDefinition = context.Variables[irn2];                                                
+                                yield break;
+                            }
+                        }
+                    }
+                    //explicit ResourceDictionary in Resources
+                    else if (resourcesNode is IElementNode resourceDictionary
+                            && resourceDictionary.XmlType.Name == "ResourceDictionary") {
+                        foreach (var rn in resourceDictionary.CollectionItems) {
+                            if (rn is IElementNode irn3
+                                && irn3.Properties.TryGetValue(XmlName.xKey, out INode xKeyNode3) 
+                                && irn3.XmlType.Name != "OnPlatform"
+                                && context.Variables.ContainsKey(irn3) 
+                                && xKeyNode3 is ValueNode xKeyValueNode3
+                                && xKeyValueNode3.Value as string == keyValueNode.Value as string)
+                            {
+                                if (irn3.CollectionItems.Count == 1 && irn3.CollectionItems[0] is ValueNode vn3 && vn3.Value is string) {
+                                    foreach (var instruction in TryConvert(vn3, eNode, vardefref, module, context))
+                                        yield return instruction;
+                                    yield break;
+                                }
+
+                                vardefref.VariableDefinition = context.Variables[irn3];                                                
+                                yield break;
+                            }
+                        } 
+                    }
+                }
+                
+                n = n.Parent as ElementNode;
+            }
+                
+
+            //Fallback
```

Comment:
Is there a chance to access the application-level resources? For example in the template app, we can't compile anything, because in the `MainPage.xaml` we reference static resources in the `Styles.xaml` where we reference static resources in `Colors.xaml`. The issue #20244 still isn't fixed with this approach if we don't support the application level styles. What if the developer moves that row definition string to a shared style resource?

Reply:
we can't assume Styles.xaml or Colors.xaml will be used, and not modified

Reply:
of course, I'm just asking if there is something we could do in that case or if the XAML files need to be always compiled separately so that pages and controls documents can't see the global application resources

Reply:
I can't think of anything